src/powerpc/ffi_darwin.c

```diff
@@ -623,37 +624,48 @@ darwin_adjust_aggregate_sizes (ffi_type *s)
 }
 
 /* Adjust the size of S to be correct for AIX.
-   Word-align double unless it is the first member of a structure.  */
+   Word-align double unless it is the first member of a structure recursively.
+   Return true if we found a recursive first member aggregate of interest. */
 
-static void
-aix_adjust_aggregate_sizes (ffi_type *s)
+static bool
+aix_adjust_aggregate_sizes (ffi_type *s, bool outer_most_type_or_first_member)
 {
   int i;
 
   if (s->type != FFI_TYPE_STRUCT)
-    return;
+    return false;
 
   s->size = 0;
+  bool nested_first_member = false;
```

Comment:
This is not strictly legal C code: declarations must be at the beginning of the block.

Reply:
Are we coding to pre-C99? `<stdbool.h>` and `_Bool` does not exist pre-C99 either.

Reply:
The compilers used by OpenJ9 tolerate this.

Reply:
libffi does not use `bool` elsewhere.  `0/1` would be more consistent with the style of libffi.

Reply:
I've adjusted the style to match and remove the bool.